frontend/test/resolution/testDomains.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright 2021-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test-resolution.h"
+
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/resolution/scope-queries.h"
+#include "chpl/types/all-types.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/Record.h"
+#include "chpl/uast/Variable.h"
+
+static std::string DomainModule =
+R"""(
+module ChapelDomain {
+  record _domain {
+    var _pid: int;
+    var _instance;
+    var _unowned:bool;
+  }
+
+  proc type _domain.static_type(param rank : int, type idxType=int, param stridable: bool = false) type {
```

Comment:
Is `static_type` intended to be user facing? I feel weird about the `_` case. Or are we just exposing it so we can test domain features?

Reply:
This is for domains only.